plugins/privacy/user/user.php

```diff
@@ -118,36 +130,52 @@ public function onPrivacyRemoveData(PrivacyTableRequest $request, JUser $user =
 
 		$user->save();
 
-		// Destroy all sessions for the user account
-		$sessionIds = $this->db->setQuery(
-			$this->db->getQuery(true)
-				->select($this->db->quoteName('session_id'))
-				->from($this->db->quoteName('#__session'))
-				->where($this->db->quoteName('userid') . ' = ' . (int) $user->id)
-		)->loadColumn();
+		// Destroy all sessions for the user account if able
+		if (!$this->app->get('session_metadata', true))
+		{
+			return;
+		}
+
+		try
+		{
+			$userId = (int) $user->id;
+
+			$sessionIds = $this->db->setQuery(
+				$this->db->getQuery(true)
+					->select($this->db->quoteName('session_id'))
+					->from($this->db->quoteName('#__session'))
+					->where($this->db->quoteName('userid') . ' = :userid')
+					->bind(':userid', $userId, ParameterType::INTEGER)
+			)->loadColumn();
+		}
+		catch (ExecutionFailureException $e)
+		{
+			return;
+		}
 
 		// If there aren't any active sessions then there's nothing to do here
 		if (empty($sessionIds))
 		{
 			return;
 		}
 
-		$storeName = Factory::getConfig()->get('session_handler', 'none');
-		$store     = JSessionStorage::getInstance($storeName);
-		$quotedIds = array();
+		/** @var SessionManager $sessionManager */
+		$sessionManager = Factory::getContainer()->get('session.manager');
 
-		// Destroy the sessions and quote the IDs to purge the session table
-		foreach ($sessionIds as $sessionId)
+		$sessionManager->destroySessions($sessionIds);
+
+		try
 		{
-			$store->destroy($sessionId);
-			$quotedIds[] = $this->db->quote($sessionId);
+			$this->db->setQuery(
+				$this->db->getQuery(true)
+					->delete($this->db->quoteName('#__session'))
+					->whereIn($this->db->quoteName('session_id'), $sessionIds, ParameterType::STRING)
```

Comment:
```suggestion
					->whereIn($this->db->quoteName('session_id'), $sessionIds, ParameterType::LARGE_OBJECT)
```
This is needed for PostgreSQL because `session_id` column has binary data type.